uiowa/uiowa
# update and reconfigure entity_usage

Resolves https://github.com/uiowa/uiowa/issues/2346

This fixes entity usage as it relates to v3 Layout Builder going forward for new add/edits while maintaining support for v2.

To fix existing, admin/config/entity-usage/batch-update or `drush entity-usage:recreate` needs to be run. This could be done on a site-by-site basis. I don't think we want to run this on all sites at once.

## Story

As an editor, I can add/edit references to other content (e.g. media) and then see that the media is being used on the page I just added/edited. Over time without developer interaction I can get a general idea of where content is being used on my website.

As an editor, if I request usage information to be updated for my website through the ITS Helpdesk, after the fact, I can see usage information for a particular piece of content throughout my entire site.

## To Test
- `composer install`
- `blt ds --site=sandbox.uiowa.edu` add `https://sandbox.prod.drupal.uiowa.edu` within stage file proxy.
- save a layout builder page with referenced media, visit one of the media items' usage tab and you should see the layout builder page you just saved.
- save a person node with tags or one that has been used as an author. on the person usage page you should see where it is being referenced as an author. on the tag term usage tab you should see the person node.


## Before/After

**before media**
<img width="770" alt="Screen Shot 2021-07-08 at 12 32 30 PM" src="https://user-images.githubusercontent.com/4663676/124966270-bbbc0400-dfe8-11eb-88da-bcc35aa3bc5d.png">

** after media **
<img width="915" alt="Screen Shot 2021-07-08 at 12 32 56 PM" src="https://user-images.githubusercontent.com/4663676/124966301-c5de0280-dfe8-11eb-8d82-70368c405e08.png">

<img width="807" alt="Screen Shot 2021-07-08 at 12 35 52 PM" src="https://user-images.githubusercontent.com/4663676/124966737-41d84a80-dfe9-11eb-8625-35a42325c9e4.png">
<img width="995" alt="Screen Shot 2021-07-08 at 12 36 22 PM" src="https://user-images.githubusercontent.com/4663676/124966741-4270e100-dfe9-11eb-959e-ec6071e69c6a.png">
<img width="734" alt="Screen Shot 2021-07-08 at 12 36 58 PM" src="https://user-images.githubusercontent.com/4663676/124966747-43a20e00-dfe9-11eb-94e4-03980a17b37e.png">
